packages/babel-plugin-proposal-private-property-in-object/src/native-private-fields.js

```diff
@@ -0,0 +1,140 @@
+import syntaxPlugin from "@babel/plugin-syntax-private-property-in-object";
+import { injectInitialization as injectConstructorInit } from "@babel/helper-create-class-features-plugin";
+
+export default function pluginPrivateIn({ types: t, template }) {
+  const classWeakSets = new WeakMap();
+  const fieldsWeakSets = new WeakMap();
+
+  function unshadow(name, targetScope, scope) {
+    while (scope !== targetScope) {
+      if (scope.hasOwnBinding(name)) scope.rename(name);
```

Comment:
I think I'll leave this as-is: renaming a class is always observable (because it changes the `.name`), while renaming the other conflicts is often not observable (if we rename a variable). It can still be observable if the conflict is with a class/function, but at least it's not always.